docarray/array/storage/redis/getsetdel.py

```diff
@@ -120,4 +118,13 @@ def _save_offset2ids(self):
         self._update_offset2ids_meta()
 
     def _clear_storage(self):
-        self._client.flushdb()
+        cursor = 0
+        while True:
```

Comment:
Can you please set `cursor=-1` and:

```while cursor != 0```

Reply:
Initial value of `cursor` has to be `0`, otherwise Redis `scan` will miss some of the mathching keys. As [Redis document](https://redis.io/commands/scan/) mentioned

> Starting an iteration with a cursor value of 0, and calling SCAN until the returned cursor is 0 again is called a full iteration.

Reply:
then add a boolean, avoid `while True`. It scares me a lot

Reply:
I have updated this function and implement it with `scan_iter` now.